java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java

```diff
@@ -0,0 +1,879 @@
+			if (platform != null) {
+				JFileChooser fileChooser = new JFileChooser(new File(".").getAbsoluteFile());
+				fileChooser.setSelectedFile(new File(tfSavePath.getText()));
+				int rv = fileChooser.showSaveDialog(pnlCenter);
+				if (rv == JFileChooser.APPROVE_OPTION) {
+					File saveFile = fileChooser.getSelectedFile();
+					tfSavePath.setText(saveFile.getAbsolutePath());
+					this.platformSaveSnapshot(platform, saveFile);
+				}
+			}
+			else {
+			}
+		});
+		pnlSave.add(tfSavePath);
+		pnlSave.add(btnSave);
+
+		JPanel pnlApply = new JPanel();
+		pnlApply.setBorder(BorderFactory.createTitledBorder("Apply Snapshot"));
+		pnlApply.setLayout(new BorderLayout(5, 5));
+		
+		JCheckBox cbCloseExistingWindows = new JCheckBox("Close Existing Windows");
+		
+		pnlApply.add(cbCloseExistingWindows, BorderLayout.NORTH);
+		
+		JPanel pnlApplyPath = new JPanel();
+		pnlApplyPath.setLayout(new BoxLayout(pnlApplyPath, BoxLayout.X_AXIS));
+		JTextField tfApplyPath = new JTextField(new File("snapshot.json").getAbsolutePath().toString());
+		tfApplyPath.setPreferredSize(new Dimension(Short.MAX_VALUE, tfApplyPath.getPreferredSize().height));
+		JButton btnApply = new JButton("Apply...");
+		btnApply.setEnabled(false);
+		btnApply.addActionListener(e -> {
+			Platform platform = this.getSelectedNode(Platform.class);
+			if (platform != null) {
+				JFileChooser fileChooser = new JFileChooser(new File(".").getAbsoluteFile());
+				fileChooser.setSelectedFile(new File(tfApplyPath.getText()));
+				int rv = fileChooser.showOpenDialog(pnlCenter);
+				if (rv == JFileChooser.APPROVE_OPTION) {
+					File snapshotFile = fileChooser.getSelectedFile();
+					tfApplyPath.setText(snapshotFile.getAbsolutePath());
+					this.platformApplySnapshot(platform, snapshotFile, cbCloseExistingWindows.isSelected());
+				}
+			}
+			else {
+			}
+		});
+		pnlApplyPath.add(tfApplyPath);
+		pnlApplyPath.add(btnApply);
+		pnlApply.add(pnlApplyPath, BorderLayout.CENTER);
+
+		pnlCenter.add(pnlSave);
+		pnlCenter.add(pnlApply);
+		pnlCenter.add(new Box.Filler(new Dimension(0, 0), new Dimension(0, Short.MAX_VALUE),
+				new Dimension(0, Short.MAX_VALUE)));
+		
+		JPanel pnlTop = new JPanel(new BorderLayout(5, 5));
+		pnlTop.setBorder(BorderFactory.createEmptyBorder(20, 10, 20, 10));
+		pnlTop.add(new JLabel("Selected Platform"), BorderLayout.WEST);
+		JTextField tfSelectedPlatformUuid = new JTextField("N/A");
+		tfSelectedPlatformUuid.setEditable(false);
+		pnlTop.add(tfSelectedPlatformUuid, BorderLayout.CENTER);
+
+		this.runtimeTree.addTreeSelectionListener(e -> {
+			Platform p = this.getSelectedNode(Platform.class);
+			btnSave.setEnabled(p != null);
+			btnApply.setEnabled(p != null);
+			tfSelectedPlatformUuid.setText(p == null ? "N/A" : p.getUuid());
+		});
+
+		pnlSnapshot.add(pnlTop, BorderLayout.NORTH);
+		pnlSnapshot.add(pnlCenter, BorderLayout.CENTER);
+		return pnlSnapshot;
+	}
+
+	JPanel createLayoutPanel() {
```

Comment:
Can we just store this in memory ?

Something like you name aLayout as you save it, goes to a list then you pick one and apply it.

Reply:
changed the implementation per comment.